Gems/PhysX/Code/Source/HingeJointComponent.cpp

```diff
@@ -85,4 +88,60 @@ namespace PhysX
             m_jointSceneOwner = leadFollowerInfo.m_followerBody->m_sceneOwner;
         }
     }
+
+    // AZ::Component
+    void HingeJointComponent::Activate() {
+        JointComponent::Activate();
+        PhysX::JointInterfaceRequestBus::Handler::BusConnect(this->GetEntityId());
+    }
+
+    void HingeJointComponent::Deactivate() {
+        JointComponent::Deactivate();
+        PhysX::JointInterfaceRequestBus::Handler::BusDisconnect();
+    }
+
+    physx::PxRevoluteJoint* GetPhysXNativeRevoluteJoint(const AzPhysics::JointHandle& joint_handle,
+                                                        const AzPhysics::SceneHandle& joint_scene_owner)
+    {
+        auto* sceneInterface = AZ::Interface<AzPhysics::SceneInterface>::Get();
+        const auto joint = sceneInterface->GetJointFromHandle(joint_handle,joint_scene_owner);
+        const auto type = joint->GetNativeType();
+        AZ_Assert(type == PhysX::NativeTypeIdentifiers::HingeJoint, "It is not PhysXHingeJoint");
+        physx::PxJoint *native_joint = static_cast<physx::PxJoint*>(joint->GetNativePointer());
+        physx::PxRevoluteJoint *native =native_joint->is<physx::PxRevoluteJoint>();
+        AZ_Assert(native, "It is not PxRevoluteJoint");
+        return native;
+    }
+
+    float HingeJointComponent::GetPosition(){
+        return GetPhysXNativeRevoluteJoint(m_jointSceneOwner,m_jointHandle)->getAngle();
+    }
+
+    float HingeJointComponent::GetVelocity() {
+        return GetPhysXNativeRevoluteJoint(m_jointSceneOwner,m_jointHandle)->getVelocity();
+    }
+
+
+    AZStd::pair<float, float> HingeJointComponent::GetLimits(){
+        auto limit = GetPhysXNativeRevoluteJoint(m_jointSceneOwner,m_jointHandle)->getLimit();
+        return AZStd::pair<float, float>(limit.lower,limit.upper);
+    }
+
+    AZ::Transform HingeJointComponent::GetTransform(){
+        auto t =GetPhysXNativeRevoluteJoint(m_jointSceneOwner,m_jointHandle)->getRelativeTransform();
```

Comment:
nit/standards: We tend to avoid single letter variable names so this would be better as `transform` (or my preferred `worldFromLocal` to show the spaces the transform is moving from/to)

Reply:
applied standard